tests/ga/test_policy_engine.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import os
+import shutil
+
+from tests.lib.tools import AgentTestCase
+from azurelinuxagent.ga.policy.policy_engine import PolicyEngine, PolicyEngineConfigurator, POLICY_SUPPORT_MATRIX
+from tests.lib.tools import patch, data_dir, test_dir
+
+
+class TestPolicyEngine(AgentTestCase):
+    patcher = None
+    regorus_dest_path = None    # Location where real regorus executable should be.
+    default_data_path = os.path.join(data_dir, 'policy', "agent-extension-default-data.json")
+    default_policy_path = os.path.join(data_dir, 'policy', "agent_extension_policy.rego")
+    test_input_path = os.path.join(data_dir, 'policy', "agent-extension-input.json")
+
+    @classmethod
+    def setUpClass(cls):
+        # Currently, ga/policy/regorus contains a dummy binary. The unit tests require a real binary,
+        # so we replace the dummy with a copy from the tests_e2e folder.
+        regorus_source_path = os.path.abspath(os.path.join(data_dir, "policy/regorus"))
+        cls.regorus_dest_path = os.path.abspath(os.path.join(test_dir, "..", "azurelinuxagent/ga/policy/regorus"))
+        if not os.path.exists(cls.regorus_dest_path):
+            shutil.copy(regorus_source_path, cls.regorus_dest_path)
+        # Patch the path to regorus for all unit tests.
+        cls.patcher = patch('azurelinuxagent.ga.policy.regorus.get_regorus_path', return_value=cls.regorus_dest_path)
```

Comment:
Add a comment explaining how binary needs to be in the agent package for real VMs, as opposed to in the egg (as it is here for unit testing purposes)